### DISCUSSION TITLE
How to render a detail view as ReadOnly with Interface (not Display)?

### DISCUSSION THREAD

**Discussion by Pachat** [directus/directus#9596] [Archive]
What I want to achieve is to show the fields of one record, not as a row in a table with many rows, 
but as read-only detail on a page with just that one record.

**Comment by benhaynes** [comment]
Could you clarify what you mean by this? Do you want items on the Collection page (Item Browse) that are not clickable?

**Comment by Pachat** [comment]
I'd like to show the values of one record, somehow like a readonly form (not as _input readonly_ which I could set overriding the CSS).
_input readonly_ would show "Status" as a readonly dropdown with "Published" (if it is published).

Instead I want to show the values as labels. And the Status as a bullet. All like it shows in a row. 
Is there something out of the box in Directus, as easy as the (wonderful) form generator, but to show readonly values of one record?

**Reply by benhaynes** [reply]
You could create a singleton collection and set the interfaces to `readonly`, but you're saying you don't want that. There's no way to set **readonly** items on the Collection Page (the listing). That could become a permission, or you could do it with a custom Layout (extension) by duplicating the Table Layout, and removing the click events.

**Comment by Pachat** [comment]
With screenshots, I hope to be more clear.
Suppose that my form is:

![form](https://user-images.githubusercontent.com/13746133/140890456-fb510f19-b543-47e6-9641-0394c6fd3c08.png)
_The form can have many fields, twenty or more_

To avoid horizontal scrolling, not all the fields will appear in the table:

![table](https://user-images.githubusercontent.com/13746133/140870194-267df18d-1b1e-4fba-92d4-0c2eb626f18d.png)


When clicking a row, how can I show it's read-only details like this?
Directus is so clever at creating forms that I was expecting such easy ways to create detail views too.


![display](https://user-images.githubusercontent.com/13746133/140891072-4382a0e1-c9a7-4b0b-b49a-a5ea4095ae35.png)

**Reply by Pachat** [reply]
How to render the Detail form just with the Interface rendering?
see https://user-images.githubusercontent.com/13746133/140891072-4382a0e1-c9a7-4b0b-b49a-a5ea4095ae35.png

**Reply by joselcvarela** [reply]
It's not possible to do that.
The only thing you can do is to create a role and give Read Permissions for that role and no other permissions. By doing that you could achieve something like this:
<img width="400" alt="Screenshot 2022-01-17 at 15 01 38" src="https://user-images.githubusercontent.com/14039341/149793214-195cf7cc-8b09-438d-978b-ff8680cd0d10.png">

You also can use shares
<img width="400" src="https://user-images.githubusercontent.com/14039341/149793697-545bccdd-0bf4-45cf-a310-3d6682f41fb8.png" />

And this is what the shareable link will look like:
<img width="400" alt="Screenshot 2022-01-17 at 15 06 25" src="https://user-images.githubusercontent.com/14039341/149793903-96c42919-accc-4c74-8912-e7e6946acd53.png">

**Reply by rijkvanzanten** [reply]
(cc @benhaynes. Could be an interesting angle to try to re-use Displays in a more dedicated readonly visual style)

**Reply by benhaynes** [reply]
Hmm, maybe, yeah. I think the form and readonly interfaces makes the most sense as of now, but we could look into an alternate view that is more "compact" and printer-friendly with only inline values.

**Reply by younky-yang** [reply]
We do need such feature to enable a collection as a readonly collection which means even admin can't edit the data. Because in some cases we also import some reference data from other systems with some ETL jobs, and these data should not edit by anyone from Directus, but we need them to show to users. 

For this purpose, to control the access from permission is kind of complex.
